js/chkRequired.js

```diff
@@ -0,0 +1,23 @@
+;(function() {
+  /**
+   * 폼 필수 input 체크
+   * - 필수 input 이 전부 빈값이면 서브밋 버튼 비활성화
+   */
+  const fmLogin = document.getElementById('form-login');
+  const btn = document.getElementById('btn-login');
+  const reqs = document.querySelectorAll('#form-login [required]');
+
+  function chkRequired () {
+    // console.log('fn:chkInput');
+
+    for (let input of reqs) {
+      if (input.value.trim() === '') {
+        btn.disabled = true;
+        return ;
+      }
+    }
+    btn.disabled = false;
+  }
+
+  fmLogin.addEventListener('input', chkRequired);
+}());
```

Comment:
잘 작성해주셨는데요! 아래처럼 변경해보는건 어떨까요!?

1. for...of 루프를 some() 메소드 사용으로 불필요한 반복을 줄일 수 있습니다.
2. DOM 요소를 찾지 못했을 경우에 대한 에러 처리를 추가했습니다.
3. change 이벤트 사용으로 불필요한 DOM 연산을 줄일 수 있습니다.

```suggestion
;(() => {
  /**
   * 폼 필수 input 체크
   * - 필수 input 이 전부 빈값이면 서브밋 버튼 비활성화
   */
  const fmLogin = document.getElementById('form-login');
  const btn = document.getElementById('btn-login');
  const reqs = document.querySelectorAll('#form-login [required]');

  if (!fmLogin || !btn || reqs.length === 0) {
    console.error('Required DOM elements not found');
    return;
  }

  const chkRequired = () => {
    btn.disabled = Array.from(reqs).some(input => input.value.trim() === '');
  };

  fmLogin.addEventListener('change', chkRequired);

  // 초기 상태 설정
  chkRequired();
})();
```